src/resolvers/git.cr

```diff
@@ -144,20 +186,47 @@ module Shards
     end
 
     def git_url
-      dependency.git.not_nil!
+      source.strip
     end
 
-    private def git_refs(version)
-      case version
+    def parse_requirement(params : Hash(String, String)) : Requirement
+      params.each do |key, value|
```

Comment:
Should we raise an `Ambiguous dependency requirement for ...` if more than one branch, tag, commit is specified instead of honoring the first one?

Reply:
Alternatively, pick the most specific one `version > tag > branch`.

Reply:
Actually a `tag` could be more specific than a `version` with a pattern. But they also could be conflicting if a version is not reachable in a branch.

If anything, I would say we could raise instead of considering just the first one. @straight-shoota do you think this would be a breaking change?

We could implement a stricter parser if we want, for the current shard.yml, but be more relaxed when the solver is scanning released versions.

Reply:
I'd say we could define this later. There is currently not defined behavior with conflicting attributes and this PR is not changing that.

Reply:
Ok. I’m out of feedback :-)

Reply:
JFTR: Shardbox database shows no dependencies with more more than one requirement restriction. So breaking things is not a concern.